src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/VaultContentXMLContentCreator.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.sling.feature.cpconverter.handlers.slinginitialcontent;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jackrabbit.JcrConstants;
+import org.apache.sling.feature.cpconverter.handlers.slinginitialcontent.xmlbuffer.XMLNode;
+import org.apache.sling.feature.cpconverter.shared.CheckedConsumer;
+import org.apache.sling.feature.cpconverter.vltpkg.JcrNamespaceRegistry;
+import org.apache.sling.feature.cpconverter.vltpkg.VaultPackageAssembler;
+import org.apache.sling.jcr.contentloader.ContentCreator;
+import org.jetbrains.annotations.NotNull;
+import javax.jcr.RepositoryException;
+import javax.jcr.Value;
+import javax.xml.stream.XMLStreamException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * ContentCreator substitute to create valid XML files to be packaged into a VaultPackage to be installed later
+ */
+public class VaultContentXMLContentCreator implements ContentCreator {
+    
+    private final String repositoryPath;
+    private final OutputStream targetOutputStream;
+    private final VaultPackageAssembler packageAssembler;
+    private final LinkedList<XMLNode> parentNodePathStack = new LinkedList<>();
+    private final JcrNamespaceRegistry namespaceRegistry;
+  
+    private final boolean isFileDescriptorEntry;
+
+    private boolean isFirstElement = true;
+    private boolean finished = false;
+    private boolean xmlProcessed = false;
+    private String primaryNodeName;
+    private XMLNode currentNode;
+    
+    public VaultContentXMLContentCreator(@NotNull String repositoryPath,
+                                         @NotNull OutputStream targetOutputStream,
+                                         @NotNull JcrNamespaceRegistry namespaceRegistry,
+                                         @NotNull VaultPackageAssembler packageAssembler,
+                                         boolean isFileDescriptorEntry) throws XMLStreamException, RepositoryException {
```

Comment:
the 2 throws clause is redundant according to sonarlint in my IDE. please remove them.